gdk4/src/texture.rs

```diff
@@ -31,3 +31,137 @@ impl<O: IsA<Texture>> TextureExtManual for O {
         }
     }
 }
+
+impl Texture {
+    pub async fn for_pixbuf_future(pixbuf: &gdk_pixbuf::Pixbuf) -> Result<Texture, glib::Error> {
+        assert_initialized_main_thread!();
+        let (sender, receiver) = futures_channel::oneshot::channel();
+
+        let task = gio::Task::new(
+            Some(pixbuf.upcast_ref()),
```

Comment:
you don't need to upcast_ref here, just `pixbuf` should be enough?

Reply:
I don't see a clear way forward for pixbufs, so ill remove that one.